packages/mdc-select/index.js

```diff
@@ -168,37 +214,78 @@ class MDCSelect extends MDCComponent {
    * on the environment's state.
    */
   initialSyncWithDOM() {
-    this.handleChange_ = () => this.foundation_.handleChange();
+    this.handleChange_ = () => this.foundation_.handleChange(/* didChange */ true);
     this.handleFocus_ = () => this.foundation_.handleFocus();
     this.handleBlur_ = () => this.foundation_.handleBlur();
-    this.handleClick_ = (evt) => this.setTransformOrigin_(evt);
+    this.handleClick_ = (evt) => this.foundation_.handleClick(this.getNormalizedXCoordinate_(evt));
+    this.handleKeydown_ = (evt) => this.foundation_.handleKeydown(evt);
+    this.handleMenuSelected_ = (evtData) => this.selectedIndex = evtData.detail.index;
+    this.handleMenuOpened_ = () => {
+      if (this.selectedIndex >= 0) {
+        this.menu_.items[this.selectedIndex].focus();
+      }
+    };
+    this.handleMenuClosed_ = () => {
+      // menuOpened_ is used to track the state of the menu opening or closing since the menu.open function
+      // will return false if the menu is still closing and this method listens to the closed event which
+      // occurs after the menu is already closed.
+      this.menuOpened_ = false;
+      if (document.activeElement !== this.selectedText_) {
+        this.foundation_.handleBlur();
+      }
+    };
+
+    const element = this.nativeControl_ ? this.nativeControl_ : this.selectedText_;
+
+    element.addEventListener('change', this.handleChange_);
+    element.addEventListener('focus', this.handleFocus_);
+    element.addEventListener('blur', this.handleBlur_);
+    element.addEventListener('keydown', this.handleKeydown_);
+
+    ['mouseup', 'touchend'].forEach((evtType) => {
```

Comment:
Listening to up events here causes 2 problems.

* This causes line ripple to behave incorrectly, since it happens before the ripple center is set. (In actual testing, it seems to come from the center on the first time, and on subsequent times I'm not seeing it ripple at all, it seems like the border is just growing vertically.)
* On the enhanced select, this triggers the menu to open as long as you release it over the select - regardless of whether the initial press happened on the select or not.

I suspect we're going to need to handle both down and up events if we go this route...

* Down would set the ripple center
* Down would set a flag that up would check... but this flag would also need to be cleared on up events anywhere else on the document to clear the flag in case the user dragged off of the select

This is becoming rather complex... IIRC the original problem is that reacting to click doesn't give the selected text time to become focused first. Would a setTimeout within the handler logic end up being simpler and allow us to continue listening to `click`?

Reply:
I'll convert these back to `mousedown`/`touchstart` and handle updating this logic in a later PR.